src/linear_algebra/determinant.lean

```diff
@@ -205,6 +205,16 @@ begin
   simpa using (linear_map.to_matrix_comp v v' v f.symm f).symm
 end
 
+/-- Specialization of `linear_equiv.is_unit_det` -/
+lemma linear_equiv.is_unit_det' {A : Type*} [integral_domain A] [module A M]
+  (f : M ≃ₗ[A] M) : is_unit (linear_map.det (f : M →ₗ[A] M)) :=
+begin
+  unfold linear_map.det,
+  split_ifs with h,
+  { convert linear_equiv.is_unit_det f _ _ }, -- `convert` avoids `decidable_eq` issues
```

Comment:
Where do these decidable_eq issues come in? This sounds like something that should be handled with a lemma closer to the problem.

Reply:
The issue is that `linear_equiv.is_unit_det` and `linear_map.det` both contain a reference to `matrix.det`, which requires a `decidable_eq` instance (in this case on `coe_sort (classical.some hS)`, so `subtype.decidable_eq` applied to a `decidable_eq M` instance). For `linear_equiv.is_unit_det` it's passed as an instance parameter, and for `linear_map.det` it's given by `classical.prop_decidable`.

I see 3 solutions here:
 * when you unfold `linear_map.det`, you need to sometimes use `convert` to make `dec_eq` coherent
 * add an extra `_def` lemma for `linear_map.det` that takes the `decidable_eq` instance as a parameter
 * change the definition of `linear_map.det` to require a `decidable_eq` instance

Reply:
Your second solution sounds like the way to go in my opinion

Reply:
I will add it to #7635 :+1: